frontend/catalyst/jax_extras/tracing.py

```diff
@@ -440,17 +440,21 @@ def deduce_signatures(
     )
 
 
-def deduce_avals(f: Callable, args, kwargs):
+def deduce_avals(f: Callable, args, kwargs, static_argnums=None):
     """Wraps the callable ``f`` into a WrappedFun container accepting collapsed flatten arguments
     and returning expanded flatten results. Calculate input abstract values and output_tree promise.
     The promise must be called after the resulting wrapped function is evaluated."""
     # TODO: deprecate in favor of `deduce_signatures`
+    wf = wrap_init(f)
+    if static_argnums:
+        argnums = [static_argnums] if isinstance(static_argnums, int) else static_argnums
```

Comment:
I think it might be in the backlog!

Reply:
We probably need to support kwargs better #814 before tackling this one?

Reply:
> Not necessary at the moment, but here it would be better to provide this validation somewhere else. Maybe just above it. Something like:
> 
> ```python
> # warning pseudocode
> def deduce_avals:
>   if # not all static argnums are integers:
>     raise appropariate error
> ```

@erick-xanadu I improved the _verify_static_argnums function to check for the correct type of static_argnums. in the case of integer (not tuple), I think compiler options take care of it, meaning that it transforms an int into a tuple with one integer in it.

Reply:
@mehrdad2m a minor comment here, but does the verification happen before we call this function? If it does then it should be guaranteed to be a tuple I think?